internal/api.go

```diff
@@ -16,34 +16,48 @@
 
 package api
 
-import (
-	"net/http"
-
-	"github.com/ChainSafe/gossamer/p2p"
-)
+// Service couples all components required for the API.
+type Service struct {
+	Api *Api
+	err <-chan error
+}
 
-// PublicP2PService offers network related RPC methods
-type PublicP2PService struct {
-	Net *p2p.Service
+// Api contains all the available modules
+type Api struct {
+	Core *coreModule
```

Comment:
Is `*coreModule` going to be comprised of all modules (Eg. author, chain, state, system [module names found here](https://polkadot.js.org/api/METHODS_RPC.html)) basically an easy way to register all modules? Otherwise, I would consider renaming

Reply:
Renamed